src/plugins/intel_cpu/src/nodes/embedding_segments_sum.cpp

```diff
@@ -122,8 +122,8 @@ void EmbeddingSegmentsSum::getIndices(int embIndex, const int*& indices, size_t&
     }
 }
 
-int32_t EmbeddingSegmentsSum::getNumSegments() const {
-    return reinterpret_cast<const int32_t *>(getParentEdgesAtPort(NUM_SEGMENTS_IDX)[0]->getMemory().GetPtr())[0];
+size_t EmbeddingSegmentsSum::getNumSegments() const {
+    return reinterpret_cast<const size_t *>(getParentEdgesAtPort(NUM_SEGMENTS_IDX)[0]->getMemory().GetPtr())[0];
```

Comment:
The same thing. Casting to `size_t*` is not correct, however the value itself can be cast to `size_t`.

Reply:
Same to up.